crates/bevy_render/src/view/mod.rs

```diff
@@ -357,9 +385,12 @@ pub fn prepare_view_uniforms(
 
 #[derive(Clone)]
 struct MainTargetTextures {
-    a: TextureView,
-    b: TextureView,
-    sampled: Option<TextureView>,
+    a: Texture,
+    a_view: TextureView,
+    b: Texture,
+    b_view: TextureView,
+    sampled: Option<Texture>,
+    sampled_view: Option<TextureView>,
```

Comment:
This invites possibly invalid state. (one is Some, the other None). I'd suggest defining a `TextureAndView` then you'd have

```rust
struct TextureAndView { texture: Texture, view: TextureView }
struct MainTargetTextures {
  a: TextureAndView,
  b: TextureAndView,
  sampled: Option<TextureAndView>,
}

Reply:
Ended up reusing `CachedTexture` which is pretty much that, and in fact it's what the methods we use to set up the texture return us, that way the setup code is also shorter/cleaner.